### DISCUSSION TITLE
How to plot a line along a pole figure

### DISCUSSION THREAD

**Discussion by matteobsu** [mtex-toolbox/mtex#806] [Ask Anything] [answered]
Hi guys,

I would like to plot a line along a pole figure.
Similar to what is done here https://mtex-toolbox.github.io/EBSDProfile.html
I was able to draw the line in the pole figure but when if I try to use the spatial_profile function it says it doesn't work for pole figures. Is there any command to do this?

Cheers,

**Comment by kilir** [comment, marked as answer]
Hi,

what would you want as a result? While the x-axis can be distance from the start of the line, what would you plot on the y-axis?

Cheers,
Rüdiger

**Reply by matteobsu** [reply]
thanks for the answer.
with respect to the attached figure I want to plot the intensity of the pole (y-axis) as function of the coordinates along the magenta line I could draw
![untitled](https://user-images.githubusercontent.com/15745811/108858961-06fee880-75ed-11eb-8fea-fe1ea4999f69.png)

**Reply by kilir** [reply]
Basically you need to generate a  bunch of vectors on your line and evaluate your polefigure at those:
Example

```
s = S2Fun.smiley; 
% use calcPDF in case you have an odf as input, respectively calcDensity if you use directions or orientations *Miller

plot(s)
mtexFig = gcm;

% select a line:
% for this function, have a look here: https://github.com/mtex-toolbox/mtex/issues/299#issuecomment-363128800
% click start and end point of line
% pos = clickRegionFromShpere(mtexFig)

% alternatively, you could of course directly define the two vectors
pos=vector3d([-0.854419 0.770532 ;  0.430298  -0.505862; 0.291226  0.387793])

% generate e.g. 100 vectors between start and end
n = cross(pos(1),pos(2)); % normal 
steps  = linspace(0,angle(pos(1),pos(2)),100) % 100 steps

rot = rotation.byAxisAngle(n,steps);
v= rot*pos(1)

%check
hold on
plot(pos,'label',{'pos1' 'pos2'},'add2all')
plot(v,'add2all')
hold off


plot(steps/degree, eval(s,v))
```
Cheers,
Rüdiger

**Comment by matteobsu** [comment]
Hi Rudiger,

Thanks. I've tried your code but it doesn't work for my data. I actually have only a PoleFigure variable from which I can plot the lattice plane pole figures such as the one above. I could also get the ODF using calcODF but when I tried the calcPDF using the h, r given in my data it prompted me that the requested array was exceeding the limit.

**Reply by kilir** [reply]
Hi,
if you want to evaluate the pole figure data, you can use `calcDensity()`, if you have an odf, `calcPDF()` should do. Maybe you need to give some more insight into what you did.

This example works without problems for me.
```
mtexdata dubna
odf = pf.calcODF
pf_c = calcPDF(odf, Miller(0,0,0,1,pf.CS))
```
Cheers,
Rüdiger

**Reply by matteobsu** [reply]
This helped!
I ended up recalculating ODF from teh pole figures, then calculating the PDF with the inclusion of the specification of the Miller indices hkl.
Thanks!
Matteo
